src/sentry/tasks/relay.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+
+import logging
+
+from django.conf import settings
+from django.core.cache import cache
+
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(name="sentry.tasks.relay.update_config_cache", queue="relay_config")
+def update_config_cache(generate, organization_id=None, project_id=None, update_reason=None):
+    """
+    Update the Redis cache for the Relay projectconfig. This task is invoked
+    whenever a project/org option has been saved or smart quotas potentially
+    caused a change in projectconfig.
+
+    Either organization_id or project_id has to be provided.
+
+    :param organization_id: The organization for which to invalidate configs.
+    :param project_id: The project for which to invalidate configs.
+    :param generate: If `True`, caches will be eagerly regenerated, not only
+        invalidated.
+    """
+
+    from sentry.models import Project
+    from sentry.relay import projectconfig_cache
+    from sentry.relay.config import get_project_config
+
+    # Delete key before generating configs such that we never have an outdated
+    # but valid cache.
+    #
+    # If this was running at the end of the task, it would be more effective
+    # against bursts of updates, but introduces a different race where an
+    # outdated cache may be used.
+    debounce_key = _get_schedule_debounce_key(project_id, organization_id)
+    cache.delete(debounce_key)
+
+    if project_id:
+        projects = Project.objects.filter(id=project_id)
```

Comment:
Please have this call and the one on L45 be cached lookups.

Reply:
✅